### PR TITLE
[New Rule] NTDS Dump via Wbadmin

### PR DESCRIPTION
**Pull request by w0rk3r**
## Issues

Part of #3005 

## Summary

Identifies the execution of wbadmin to access the NTDS.dit file in a domain controller. Attackers with privileges from groups like Backup Operators can abuse the utility to perform credential access and compromise the domain.

### CODE REVIEW

File: rules/windows/credential_access_wbadmin_ntds.toml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+creation_date = "2024/06/05"
+integration = ["windows", "endpoint", "system"]
+maturity = "production"
+updated_date = "2024/06/05"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies the execution of wbadmin to access the NTDS.dit file in a domain controller. Attackers with privileges from
+groups like Backup Operators can abuse the utility to perform credential access and compromise the domain.
```
Comment (Aegrah) [inline rules/windows/credential_access_wbadmin_ntds.toml:11]: Do these groups have common IDs that can be added to the query?

**Review by Aegrah — APPROVED** [review]

File: rules/windows/credential_access_wbadmin_ntds.toml (hunk 1 above)
Comment (w0rk3r) [reply rules/windows/credential_access_wbadmin_ntds.toml:11]: Process events don't contain the groups that the user is a part of, but to be able to execute this action, the user needs to have the SeBackupPrivilege right assigned

File: rules/windows/credential_access_wbadmin_ntds.toml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    "Tactic: Credential Access",
+    "Data Source: Elastic Endgame",
+    "Data Source: Elastic Defend"
+]
+timestamp_override = "event.ingested"
+type = "eql"
+
+query = '''
+process where host.os.type == "windows" and event.type == "start" and
+    process.name : "wbadmin.exe" and process.args : "recovery" and process.args : "*ntds.dit*"
```
Comment (w0rk3r) [inline rules/windows/credential_access_wbadmin_ntds.toml:43]:
```suggestion
any where host.os.type == "windows" and event.category : ("process", "file") and 
    event.type in ("start", "creation") and
    (
        (process.name : "wbadmin.exe" and process.args : "recovery" and process.args : "*ntds.dit*") or
        (process.name : "wbengine.exe" and file.name : "ntds.dit")
    )
```

We can also do something like this

Comment (Samirbous) [inline rules/windows/credential_access_wbadmin_ntds.toml:43]: for performance maybe just we limit to process creation (the proc will be logged first anyway):
```suggestion
process where host.os.type == "windows" and event.type == "start" and
    (process.name : "wbadmin.exe" or ?process.pe.original_file_name : "wbadmin.exe") and 
     process.args : "recovery" and process.command_line : "*ntds.dit*"
```

**Review by Samirbous — APPROVED** [review]

**Review by terrancedejesus — APPROVED** [review]
Great addition!

**Pull request merged**